src/wallet/wallet.h

```diff
@@ -381,7 +381,7 @@ struct SaplingNoteEntry
     SaplingOutPoint op;
     libzcash::SaplingPaymentAddress address;
     libzcash::SaplingNote note;
-    std::array<unsigned char, ZC_MEMO_SIZE> memo;
+    std::optional<libzcash::Memo> memo;
```

Comment:
So, the updated `Memo` class can no longer hold a `0xF600…00` value. `std::optional<Memo>` means “a possibly-empty memo”.

Prior to this PR, both `std::nullopt` and `std::optional(Memo({{0xf6}}))` were representations of the empty memo, but they weren’t always checked consistently. And there were multiple constants defined as `{{0xf6}}`, as well as other places that `{{0xf6}}` was used directly as a magic array. So, a significant improvment in this PR is that `std::nullopt` is the only empty memo representation.

tl;dr – if we just had `libzcash::Memo memo;` here, then that wouldn’t allow us to represent the empty memo.

Reply:
No; I'm suggesting that this be `Memo::Bytes`.

Reply:
Ah, I see … IMO, we should stick with the “higher-level” wrapper as much as possible. Changing `SaplingNoteEntry` to  use `Bytes` leaks to the RPC layer. `BaseNotePlaintext` is the only thing that deals with serialization/encryption, so that’s where we use `Memo::Bytes`.